### PR TITLE
Improve time savings calculation math for test reordering

### PR DESCRIPTION
**Pull request by ZainRizvi**
Use a more accurate method that accounts for tests being run in parallel

Right now we still log results to the console, but later it'll get logged to Rockset for better tracking

### CODE REVIEW

**Review by huydhn — APPROVED** [review]
LGTM!

File: tools/testing/test_selections.py [hunk 1]
```diff
@@ -195,6 +196,67 @@ def _python_test_file_to_test_name(tests: Set[str]) -> Set[str]:
...
+    max_time_savings_sec = 0.0
+
+    # It's easier to look up prioritized tests by name
+    prioritized_test_names = {test.name for test in prioritized_tests}
+
+    for test in selected_tests:
+        serial = is_serial_test_fn(test.name)
+        if test.name in prioritized_test_names:
+            # Successive tests will always have a greater time savings
+            max_time_savings_sec = default_pool.next_test_start_time(
```
Comment (huydhn) [inline tools/testing/test_selections.py:242]: Should there also be a max here as the name implies, i.e. max_time_savings_sec = max(max_time_savings_sec, default_pool.next_test_start_time(serial) - prioritized_pool.next_test_start_time(serial)`

Comment (huydhn) [reply tools/testing/test_selections.py:242]: Got it!  I should have pay more attention to comments when reviewing code.  I try to read the code to understand it as it goes, then map my review questions accordingly.  I didn't even notice the comment until you point it out explicitly (blind spot, lol)

**Review by malfet — APPROVED** [review]

File: tools/test/test_test_selections.py [hunk 2]
```diff
@@ -412,6 +417,99 @@ def test_get_reordered_tests(
...
+
+        expected_time_savings = 9.0
+
+        time_savings = log_time_savings(
+            tests, prioritized_tests, is_serial_test_fn=never_serial, num_procs=2
+        )
+        self.assertEqual(
+            time_savings, expected_time_savings, "Received an unexpected time savings"
+        )
+        pass
```
Comment (malfet) [inline tools/test/test_test_selections.py:441]: Nit
```suggestion
```

**Pull request closed without merging**